rules/0230-CSharp.md

```diff
@@ -1 +1,121 @@
-<!--- this is only a placeholder -->
+## CSharp
+CSharp is the only language that doesn't have a proper documentation on Codewars
+currently. This will probably change at some point. The test framework is
+[NUnit](http://www.nunit.org/), initially a port from JUnit. It's rather large,
+so we cannot cover [everything][Nunit doc] here, but at least some points
+should be addressed.
+
+*Note:* C# looks rather ugly in the generated PDF, so I'm going to refer to it
+as _CSharp_ instead.
+
+ [Nunit doc]: http://www.nunit.org/index.php?p=docHome&r=2.6.4
+
+### Beware of the assert arguments!
+NUnit differs from the other frameworks. Very. Much. Well, overall it doesn't,
+it still has it's `Assert.AreEqual` and `Assert.Greater` and others. __But__ the
+order of arguments is swapped.
+
+Where CoffeeScript, Ruby, Python and Haskell expect the first argument of
+`Test.assert_equals`/`Test.assertEquals`/`shouldBe` to be the actual value and
+the second to be the expected, NUnit swaps the order, see the documentation of
+[Assert]:
+
+``` charp
+Assert.AreEqual( int expected, int actual );
+Assert.AreEqual( int expected, int actual, string message );
+Assert.AreEqual( int expected, int actual, string message,
+                 params object[] parms );
+// many other
+```
+
+For comparison, here's the syntax of an equality assert in the other languages:
+
+``` haskell
+-- Haskell
+actual `shouldBe` expected
+```
+``` javascript
+// Javascript
+Test.assertEquals(actual, expected, message);
+```
+``` python
+# Python
+Test.assert_equals(actual, expected, message)
+```
+``` ruby
+# Ruby
+Test.assert_equals(actual, expected, message)
+```
+
+Keep this in mind. Your tests won't fail if your swap the arguments, but they
+__will__ confuse anyone who fails to pass your assertion.
+
+ [Assert]: http://www.nunit.org/index.php?p=equalityAsserts&r=2.6.4
+
+### Use descriptive test names
+This was already addressed in the previous section, but it's rather important
+in CSharp. NUnit doesn't give you any method to label a test. Instead, a test
+is simply a public static method with a `[Test]` attribute inside a class with
+the `[TestFixture]` attribute. The _name_ of your test is the method's name.
+
+Make that name descriptive. A method called `test1` doesn't help.
+
+### Use `[TestCase]` for multiple test cases
```

Comment:
I only came across `[TestCase]` by chance. @Killfile posted [a comment](http://www.codewars.com/kata/559590633066759614000063/discuss#55c505765d0910da98000120) and @g964's comment made me have a look at NUnit's documentation.